packages/framework-types/src/provider.ts

```diff
@@ -144,6 +144,15 @@ export interface ProviderEventsLibrary {
     snapshotEnvelope: NonPersistedEntitySnapshotEnvelope,
     config: BoosterConfig
   ): Promise<EntitySnapshotEnvelope>
+
+  /**
+   * Stores events envelopes that have been dispatched in the dispatched events table
+   *
+   * @param eventEnvelope - The `Array<EventEnvelope>` to store
+   * @param config - The Booster configuration object
+   * @returns `true` if the dispatched event was stored, otherwise, returns `false`
```

Comment:
I would be more restrictive and explicit here and say:
```suggestion
   * @returns `true` if the dispatched event was stored, `false` if the event already exists, and will throw in case of any other error
```